### ISSUE TITLE
Include ‘Page title’ to search results

### ISSUE THREAD

**Issue by paulhibbitts** [docsifyjs/docsify#2475]
#### Problem or desire

When viewing the search results, the key context of which page a search match is found is not included. This often means one or more search result items need to be clicked to view this information to assess the suitability of the displayed match.

#### Proposal

In addition to relevant Header and match, include the page title (constructed from the filename of each match or perhaps the first H1 header?) You can see an example of this technique (and the value added by including Page title) on https://docsify-this.net/ by searching for ‘usability’ or ‘slides’ or ‘design’  through the link https://docsify-this.net/?basePath=https://raw.githubusercontent.com/hibbitts-design/docsify-this-multiple-page-course-site/main&homepage=home.md&sidebar=true&loadSidebar=_sidebar.md&name=CPT-363&hide-credits=true&browser-tab-title=CPT-363&search=true#/

Please note due to technical challenges I was not able to display the Page title first, then relevant header (I could only figure out how to display Page title under the existing related Header).

Without development constraints, I would have preferred to use this search result presentation:
### Search result closest Header (as is now)
<search result text and nearby text if applicable, with search text highlighted> (as is now)

**Comment by hiSandog** [comment]
Agreed — the page title (first H1, falling back to a title-cased filename) is the missing context that makes search results actionable without clicking through. Implementation-wise, the search plugin currently indexes per matched header/section and the result item shows that header; to show the page title the index would need to carry the page's first H1 alongside each matched section, and dedupe when the matched header *is* the page H1 (so it isn't rendered twice).

A couple of edges to design for: long page titles need truncation (and RTL handling, given the recent RTL work), and ordering — your preferred `Page Title` above `Matched Header` differs from today's layout, so it'd want to be opt-in (e.g. a `search.showPageTitleInSearch` option or a render hook) rather than changing default UX for everyone. The search plugin is the right place to scope this so the core stays untouched.

**Comment by paulhibbitts** [comment]
Thanks for your thoughts @hiSandog. If you are interested, I 'hacked' together a custom search plugin for Docsify v5 which you are welcome to use in the meanwhile - you can see it in action with a multiple page site with my Docsify-This preview using Docsify v5:
https://preview-v2.docsify-this.net/?basePath=https://raw.githubusercontent.com/hibbitts-design/docsify-this-multiple-page-open-publishing-site/main&homepage=home.md&sidebar=true&loadSidebar=_sidebar.md&searchbox=true&hide-credits=true#/

GitHub repo code:
https://github.com/paulhibbitts/test-docsify-this/blob/test-docsify-this-v2/docs/assets/vendor/docsify/plugins/custom/search.min.js

**Comment by sy-records** [comment]
Fixed via #2767
